lib/web_ui/lib/src/engine/html/shaders/shader.dart

```diff
@@ -514,46 +546,102 @@ class GradientRadial extends EngineGradient {
     ShaderDeclaration fragColor = builder.fragmentColor;
     ShaderMethod method = builder.addMethod('main');
     // Sweep gradient
-    method.addStatement(
-        'vec2 center = 0.5 * (u_resolution + u_tile_offset);');
+    method.addStatement('vec2 center = 0.5 * (u_resolution + u_tile_offset);');
     method.addStatement(
         'vec4 localCoord = vec4(gl_FragCoord.x - center.x, center.y - gl_FragCoord.y, 0, 1) * m_gradient;');
-    method.addStatement(
-        'float dist = length(localCoord);');
+    method.addStatement('float dist = length(localCoord);');
     method.addStatement(''
         'float st = abs(dist / u_radius);');
     final String probeName =
-      _writeSharedGradientShader(builder, method, gradient, tileMode);
+        _writeSharedGradientShader(builder, method, gradient, tileMode);
     method.addStatement('${fragColor.name} = ${probeName} * scale + bias;');
     String shader = builder.build();
     return shader;
   }
 }
 
-class GradientConical extends EngineGradient {
-  GradientConical(this.focal, this.focalRadius, this.center, this.radius,
-      this.colors, this.colorStops, this.tileMode, this.matrix4)
-      : super._();
+/// TODO: Implement focal.
+class GradientConical extends GradientRadial {
+  GradientConical(
+      this.focal,
+      this.focalRadius,
+      ui.Offset center,
+      double radius,
+      List<ui.Color> colors,
+      List<double>? colorStops,
+      ui.TileMode tileMode,
+      Float32List? matrix4)
+      : super(center, radius, colors, colorStops, tileMode, matrix4);
 
   final ui.Offset focal;
   final double focalRadius;
-  final ui.Offset center;
-  final double radius;
-  final List<ui.Color> colors;
-  final List<double>? colorStops;
-  final ui.TileMode tileMode;
-  final Float32List? matrix4;
 
   @override
   Object createPaintStyle(html.CanvasRenderingContext2D? ctx,
       ui.Rect? shaderBounds, double density) {
-    throw UnimplementedError();
+    if ((tileMode == ui.TileMode.clamp || tileMode == ui.TileMode.decal) &&
+        focalRadius == 0.0 &&
+        focal == const ui.Offset(0, 0)) {
+      return _createCanvasGradient(ctx, shaderBounds, density);
+    } else {
+      initWebGl();
+      return _createGlGradient(ctx, shaderBounds, density);
+    }
   }
 
   @override
-  Object createImageBitmap(ui.Rect? shaderBounds, double density,
-      bool createDataUrl) {
-    throw UnimplementedError();
+  String _createRadialFragmentShader(
+      NormalizedGradient gradient, ui.Rect shaderBounds, ui.TileMode tileMode) {
```

Comment:
Is this covered by tests?

Reply:
No. I added issue to comment https://github.com/flutter/flutter/issues/76643. focal at 0,0 is covered by existing radial tests. Once we have full 2 point radials, we'll add goldens.

Reply:
SGTM